### PR TITLE
miscellaneous lint, clean-up and modernization changes to webdriver: part II

### PR DESCRIPTION
**Pull request by thiagowfx**
Follow-up-of: https://github.com/web-platform-tests/wpt/pull/40470

### CODE REVIEW

**Review by jrandolf — APPROVED** [review]

File: webdriver/tests/bidi/__init__.py
Comment (whimboo) [inline webdriver/tests/bidi/__init__.py:74]: I don't like that we only partly use typing. Can we actually do it all together or not at all?

Comment (thiagowfx) [reply webdriver/tests/bidi/__init__.py:74]: I couldn't figure out what the type of BiDi session is.

Comment (thiagowfx) [reply webdriver/tests/bidi/__init__.py:74]: Fair enough! I missed `text:`. Fixed. Let me know if you find any other missing arguments.

File: tools/webdriver/webdriver/bidi/modules/input.py
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/input.py:4]: Why this change? It's actually the opposite what you did in line 129. And I think this change is no correct.

File: tools/webdriver/webdriver/bidi/modules/input.py
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/input.py:324]: Why the different indentation here compared to the others above? Is that what mypy suggests? Maybe because of the `origin` line being too long? I mean I like the 4 char indentation and not the 8 char one, but would like to know why it's different.

File: tools/webdriver/webdriver/bidi/modules/input.py
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/input.py:390]: For consistency?

```suggestion
    def perform_actions(self, 
                                        actions: Union[Actions, List[Any]],
                                        context: str) -> Mapping[str, Any]:
```

File: webdriver/tests/bidi/__init__.py
Comment (whimboo) [inline webdriver/tests/bidi/__init__.py:90]: Why did you remove the documentation. It's still helpful because of the extra info.

File: webdriver/tests/bidi/input/perform_actions/key_events.py
Comment (whimboo) [inline webdriver/tests/bidi/input/perform_actions/key_events.py:25]: Isn't the following better or does the linter complain? We use it quite often.

```suggestion
async def test_non_printable_key_sends_events(
    bidi_session, top_context, key, event
):
```

File: webdriver/tests/bidi/input/perform_actions/pointer_origin.py
Comment (whimboo) [inline webdriver/tests/bidi/input/perform_actions/pointer_origin.py:43]: What's the difference to line 53 for being formatted differently?

**Review by whimboo — CHANGES_REQUESTED** [review]

**Comment by thiagowfx** [comment]
All the formatting changes were done by `yapf`. The explanation for most of your comments is: That's because `yapf` did so.
Formatting wasn't the main goal, it was just a side effect from saving the file (I had auto formatting set to ON in VSCode).

If you're happy with all changes except formatting, then I can revert all formatting changes one by one and republish the PR. But it's probably not worth to go into detail in each formatting change.

Either we accept everything as is, or I revert all formatting changes and resubmit.

I'll revert the ones you commented in.

File: webdriver/tests/bidi/__init__.py
Comment (thiagowfx) [reply webdriver/tests/bidi/__init__.py:90]: It adds no value.

- bidi_session is the name of the variable, just being repeated
- the return is the name of the function

Arguable the only piece of information the documentation provides is "context" = "browsing context id". I would have renamed the variable to browsing_context_id but for the sake of consistency (we use "context" everywhere) I did not.

Are you sure you want to add the docs back?

File: webdriver/tests/bidi/input/perform_actions/pointer_origin.py
Comment (thiagowfx) [reply webdriver/tests/bidi/input/perform_actions/pointer_origin.py:43]: Because it exceed 80 lines

**Comment by whimboo** [comment]
> All the formatting changes were done by `yapf`. The explanation for most of your comments is: That's because `yapf` did so. Formatting wasn't the main goal, it was just a side effect from saving the file (I had auto formatting set to ON in VSCode).

Note that we never agreed on a formatting and there are several different tools out there all with a different opinion. For code under `tools/` we use `mypy` and `yapf` is not a thing at all in this repository as I can see. I'll let @jgraham comment here in what we actually want to have and continue. I've to second as well that discussions like that shouldn't be necessary if we would have proper linting for tests as well.

File: webdriver/tests/bidi/__init__.py
Comment (whimboo) [reply webdriver/tests/bidi/__init__.py:90]: Yeah you are right. We can leave it out.

**Review by whimboo — APPROVED** [review]

**Comment by whimboo** [comment]
Azure jobs are failing. I've re-triggered those jobs to see if it was just flakiness.

**Comment by thiagowfx** [comment]
> Azure jobs are failing.
> I've re-triggered those jobs to see if it was just flakiness.

Thanks. Indeed, we're used to flakiness in the `-stability` jobs, but not in Azure, so that's a good call.
It seems they are passing now.

**Comment by thiagowfx** [comment]
@jgraham @gsnedders please merge when you have a chance (sorry for the ping, trying to remove a need for it in https://github.com/web-platform-tests/wpt/issues/40860).

File: webdriver/tests/bidi/script/call_function/result.py
Comment (whimboo) [inline webdriver/tests/bidi/script/call_function/result.py:120]: @thiagowfx I assume this reformatting wasn't intended to get pushed?

Comment (whimboo) [reply webdriver/tests/bidi/script/call_function/result.py:120]: Especially I want to mention that we should not push further changes once a PR has been reviewed and is waiting for getting merged.

Comment (thiagowfx) [reply webdriver/tests/bidi/script/call_function/result.py:120]: Done (now it's consistent)

**Pull request merged**
